woocommerce/woocommerce-gutenberg-products-block
# Remove custom `Icon` component in favour of `@wordpress/icons` where possible

This PR removes our `Icon` component in favour of the Icon component that is included with `@wordpress/icons`. On top of this, I've removed/migrated as many `@woocommerce/icons` as possible in favour of similar or equivalent `@wordpress/icons` in order to reduce our package size.

Fixes #3512

List of substitutions: 

| Woo Library | Usage | WP Library | 
|:---:|:-----:|:---:|
| atom | Product Elements/Atomic Blocks Section | woo |
| done | Cart button checkmark | check |
| folder | Block icon | file or archive |
| image | Block icon | image |
| star | Block icon | starEmpty |
| tag | Block icon | tag |
| grid | Block icon | grid |
| bill | Block icon | currencyDollar |
| box | Block icon | box |
| notes | Block icon | page |
| bookmark | Block icon | heading |
| arrowBack | Button | arrowLeft |
| noAlt | Chip component | closeSmall |
| card | Gateways | payment |
| bank | Gateways | institution |
| search | Search components | search |
| notice | Error message components | warning |
| discussion | Reviews widgets | postComments |
| chevronUp/chevronDown | Panels | chevronUp/chevronDown |
| external | External links in panels | external |
| server | Block icon for attributes | category |
| truck | Block icon | shipping |
| stonks | Block icon | trendingUp |
| address | Block icon | mapMarker |
| comment | Block icon | commentContent |
| asterisk | Terms and Conditions Block | customPostType |
| list | Block icon | list |
| review | Block icon | commentContent |
| reader | Single product block icon | mediaAndText |
| restore | Single Product Block | backup |
| contact | Contact Block Icon | atSymbol |
| tags | Products by attribute Block | category |
| more | Products by tag block | tag |
| exclamation | New products block icon | sparkles |
| widgets | Handpicked products block | category |

#### Other Checks

- [ ] We need to confirm the packages are available in supported WordPress versions before shipping
- [x] Need to confirm bundle size has not increased too much on frontend where icons are used

### Screenshots

| Before | After |
| ------------- | ------------- |
| <img width="248" alt="Screenshot 2022-02-01 at 12 36 17" src="https://user-images.githubusercontent.com/1628454/151965347-e6802d2f-6cb2-463a-8c7b-76b0594d8210.png"> |  <img width="250" alt="Screenshot 2022-02-01 at 13 04 10" src="https://user-images.githubusercontent.com/1628454/151971955-8907a318-b461-45ae-b586-e7db9b650284.png"> |
| <img width="251" alt="Screenshot 2022-02-01 at 13 06 50" src="https://user-images.githubusercontent.com/1628454/151965729-2dde8265-d323-4376-b392-de794bc50098.png"> |  <img width="252" alt="Screenshot 2022-02-01 at 13 05 45" src="https://user-images.githubusercontent.com/1628454/151965707-3b6141df-a83b-483d-9792-18d20347ea11.png"> |

### Testing

### Manual Testing

**Confirm there are no build errors** since this should pick up any mistakes with icon naming.

How to test the changes in this Pull Request:

1. Open block inserted and smoke test block icons
2. Try inner blocks within cart/checkout/single-product/all product blocks and confirm icons are working for inner blocks.
3. Check storybook works for icon library

No testing steps for end-users.

### Performance Impact

We need to confirm bundles sizes are not increased due to using the `@wordpress/icons` package. 

### Changelog

> Update block icons to use `@wordpress/icons` where possible.
